Bump securedrop-grsec-focal metapackage to 5.4.88

emkll opened the pull request:
## Status

Ready for review
## Description of Changes

Fixes #5479 and towards https://github.com/freedomofpress/securedrop/issues/4768

Updates the kernel metapackage version to 5.4.88 for Focal installs.

See https://github.com/freedomofpress/securedrop-dev-packages-lfs/pull/89 

## Testing

- [ ] Focal metapackage pulls in 5.4.88 kernel image
- [ ] Xenial metapackage pulls in 4.14.188 kernel image
## Deployment

For new and existing installs, these changes will be distributed via deb packages through apt.

## Checklist

### If you made changes to the server application code:

- [ ] Linting (`make lint`) and tests (`make test`) pass in the development container

### If you made changes to `securedrop-admin`:

- [ ] Linting and tests (`make -C admin test`) pass in the admin development container

### If you made changes to the system configuration:

- [ ] [Configuration tests](https://docs.securedrop.org/en/latest/development/testing_configuration_tests.html) pass

### If you added or removed a file deployed with the application:

- [ ] I have updated AppArmor rules to include the change

### If you made non-trivial code changes:

- [x] I have written a test plan and validated it for this PR

emkll committed:
Bump securedrop-grsec-focal metapackage to 5.4.88

This will pull in and install 5.4 series kernels for Focal installs,
thanks to the split metapackage logic introduced in #5691

emkll committed:
Fix test for Focal kernel version

conorsch commented on molecule/testinfra/common/test_grsecurity.py:
```diff
@@ -4,7 +4,10 @@
 
 sdvars = testutils.securedrop_test_vars
-KERNEL_VERSION = sdvars.grsec_version
+if host.system_info.codename == "xenial":
```
`host` isn't available outside function scope, will need a refactor.

kushaldas committed:
Fixes the grsec test with python reformatting

KERNEL_VERSION is defined for each function as required.

kushaldas reviewed (dismissed):
I have updated the test code to make sure it goes through CI.

Also verified that: 

- [x] Focal metapackage pulls in 5.4.88 kernel image
- [x] Xenial metapackage pulls in 4.14.188 kernel image

But, the `IRDA` tests are failing for the kernel: https://app.circleci.com/pipelines/github/freedomofpress/securedrop/1812/workflows/411763ec-cf39-43d3-a77b-ab89388a6b36/jobs/49865 @emkll what do you think?

emkll committed:
Fix IRDA test for 5.4.x kernel

CONFIG_IRDA is simply not present in the kernel configuration for 5.4 series kernels, in previous kernel configurations it was commented out.

emkll commented:
There's currently an issue with the split logic for the config and grsec metapackages (introduced in #5684 and #5691 respectively do not produce the expected package contents due to filepath not being the same as the package name, resulting in some unwanted subfolders within the package:

```
total 36K
drwxr-xr-x  5 user    user    4.0K Feb  5 14:57 .
drwxrwxrwt 29 root root  20K Feb  8 09:56 ..
drwxr-xr-x  3 user    user    4.0K Feb  5 14:57 etc
drwxr-xr-x  3 user    user    4.0K Feb  5 14:57 -focal
drwxr-xr-x  3 user    user    4.0K Feb  5 14:57 opt
```

conorsch commented:
Looks like that extraneous dir only appears in the Focal metapackage, not in the Xenial one. It's empty, but should certainly be removed. Digging a bit now.

Commit:
Fix Focal kernel metapackage

Discovered by @emkll in [0]. The forked package build logic was
including an empty dir based on the distro string. Update the path
munging logic to use the proper var to avoid.

[0] https://github.com/freedomofpress/securedrop/pull/5772#issuecomment-775208976

conorsch commented:
@emkll I pushed a fix that appears to resolve for both Focal & Xenial.

conorsch commented:
@kushaldas This looks good to my eye, any concerns with merging?

rmol commented:
I have an alternative that simplifies the distribution branching that I've been hoping to share for review. It's just been hard to test because of all the other Focal problems on develop right now. Fingers crossed it'll be ready for a look by midday, and obviously I think it's worth trying to dig out from under this a bit now, but if you'd rather just cross this one off and move on, please go ahead and merge, and maybe we can clean it up later, in any event when we remove the Xenial logic.

conorsch commented:
Sounds great, @rmol, I can review later today. Happy to wait until then. Regarding keeping the distribution branching clean, I had similar thoughts on https://github.com/freedomofpress/securedrop/pull/5780, so perhaps we can reuse the approach you propose in a few places.

kushaldas commented:
> @kushaldas This looks good to my eye, any concerns with merging?

Nope, I think it is good to merge when you want to.

conorsch commented:
> if you'd rather just cross this one off and move on, please go ahead and merge, and maybe we can clean it up later

Indeed, some folks are blocked on Focal hardware installs, so proceeding with merge. Happy to revisit any additional logic you've got queued up, @rmol, just point me to a branch!
